libs/components/forms/src/lib/modules/character-counter/character-counter.component.spec.ts

```diff
@@ -155,6 +168,111 @@ describe('Character Counter component', () => {
       expect(characterCountLabelLastName.innerText.trim()).toBe('4/5');
     });
 
+    it('should announce to screen readers every 10 characters when within 50 characters of the limit', fakeAsync(() => {
+      // Sets the screen reader to the initial state
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 5');
+      component.setCharacterCountLimit(49);
+      fixture.detectChanges();
+
+      setInputValue(fixture, '1'.repeat(9));
+      // Sets currently typed characters do not change until a breakpoint
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 49');
+
+      setInputValue(fixture, '1'.repeat(10));
+      expect(getScreenReaderText(fixture)).toBe('10 characters out of 49');
+
+      setInputValue(fixture, '');
+      expect(getScreenReaderText(fixture)).toBe('0 characters out of 49');
+    }));
+
+    it('should announce to screen readers every 50 characters when not within 50 characters of the limit', fakeAsync(() => {
+      // Sets the screen reader to the initial state
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 5');
+      component.setCharacterCountLimit(99);
+      fixture.detectChanges();
+
+      setInputValue(fixture, '1'.repeat(9));
+      // Sets currently typed characters do not change until a breakpoint
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 99');
+
+      setInputValue(fixture, '1'.repeat(10));
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 99');
+
+      // Should not update when 50 characters is hit on a non-multiple of 10
+      setInputValue(fixture, '1'.repeat(49));
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 99');
+
+      setInputValue(fixture, '1'.repeat(50));
+      expect(getScreenReaderText(fixture)).toBe('50 characters out of 99');
+
+      setInputValue(fixture, '1'.repeat(60));
+      expect(getScreenReaderText(fixture)).toBe('60 characters out of 99');
+
+      setInputValue(fixture, '');
+      expect(getScreenReaderText(fixture)).toBe('0 characters out of 99');
+    }));
+
+    it('should announce to screen readers when backspacing at breakpoints', fakeAsync(() => {
+      // Sets the screen reader to the initial state
+      expect(getScreenReaderText(fixture)).toBe('4 characters out of 5');
+      component.setCharacterCountLimit(99);
+      fixture.detectChanges();
+
+      setInputValue(fixture, '1'.repeat(60));
+      expect(getScreenReaderText(fixture)).toBe('60 characters out of 99');
```

Comment:
This pair of statements is repeated many times in this set of tests, only differing by the number of repeats. This can be pulled out into a function.

```
function validateScreenReaderText(fixture: ComponentFixture<unknown>, characterCount: number): void {
  setInputValue(fixture, '1'.repeat(characterCount));
  expect(getScreenReaderText((fixture)).toBe(`${characterCount} characters out of 99`);
}
```

Reply:
Done